public/store/DetectorsStore.tsx

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import React from 'react';
+import { DetectorsService } from '../services';
+import { NotificationsStart } from 'opensearch-dashboards/public';
+import { CreateDetectorState } from '../pages/CreateDetector/containers/CreateDetector';
+import { ICalloutProps, resolveType, TCalloutColor } from '../pages/Main/components/Callout';
+import { CreateDetectorResponse, ISavedObjectsService, ServerResponse } from '../../types';
+import { CreateMappingsResponse } from '../../server/models/interfaces';
+import { logTypesWithDashboards, ROUTES } from '../utils/constants';
+import { EuiButton, EuiFlexGroup, EuiFlexItem } from '@elastic/eui';
+import { Toast } from '@opensearch-project/oui/src/eui_components/toast/global_toast_list';
+import { RouteComponentProps } from 'react-router-dom';
+import { DataStore } from './DataStore';
+import { v4 as uuidv4 } from 'uuid';
+
+export interface IDetectorsStore {
+  readonly service: DetectorsService;
+  readonly notifications: NotificationsStart;
+  readonly savedObjectsService: ISavedObjectsService;
+  readonly history: RouteComponentProps['history'] | undefined;
+  setState: (state: IDetectorsState, history: RouteComponentProps['history']) => void;
+  getState: () => IDetectorsState | undefined;
+  deleteState: () => void;
+  getPendingState: () => Promise<{
+    detectorId?: string;
+    dashboardId?: string;
+    ok: boolean;
+  }>;
+  setHandlers: (
+    calloutHandler: (callout?: ICalloutProps) => void,
+    toastHandler: (toasts?: Toast[]) => void
+  ) => void;
+}
+
+export interface IDetectorsCache {}
+
+export interface IDetectorsState {
+  pendingRequests: Promise<any>[];
+  detectorState: CreateDetectorState;
```

Comment:
I would suggest calling this just `detector` or something like `detectorInput` since it is inside `IDetectorState`

Reply:
yes, done